hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfspp/CMakeLists.txt

```diff
@@ -53,30 +53,14 @@ find_package(Threads)
 include(CheckCXXSourceCompiles)
 include(CheckSymbolExists)
 
-# Download and build gtest
-configure_file(CMakeLists-gtest.txt.in googletest-download/CMakeLists.txt)
-execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" .
-  RESULT_VARIABLE result
-  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/googletest-download)
-if(result)
-  message(FATAL_ERROR "CMake step for googletest failed: ${result}")
-endif()
-execute_process(COMMAND ${CMAKE_COMMAND} --build .
-  RESULT_VARIABLE result
-  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/googletest-download)
-if(result)
-  message(FATAL_ERROR "Build step for googletest failed: ${result}")
-endif()
-
-# Prevent overriding the parent project's compiler/linker
-# settings on Windows
+include(FetchContent)
+FetchContent_Declare(
+  googletest
+  URL https://github.com/google/googletest/archive/703bd9caab50b139428cea1aaff9974ebee5742e.zip
```

Comment:
How safe is this location?

Reply:
This location points to a location in the googletest repo, it should be safe. The SHA `703bd9caab50b139428cea1aaff9974ebee5742e` corresponds to the `release-1.10.0` tag.

Reply:
Also, the gtest repo suggests has mentioned this approach in their README.md - https://github.com/google/googletest/tree/f2d71f0be2ff2230789cfaf37d4c4f38db5b1c7c/googletest#incorporating-into-an-existing-cmake-project.